tests/amqpconnection_ssl_by_cert.phpt

```diff
@@ -0,0 +1,156 @@
+--TEST--
+AMQPConnection - ssl support / authentication using certificates
+--SKIPIF--
+<?php if (!extension_loaded("amqp")) print "skip"; ?>
+--FILE--
+<?php
+/*
+openssl s_server -accept 8443 \
+-cert /home/vagrant/php-amqp/provision/test_certs/server/cert.pem \
+-key /home/vagrant/php-amqp/provision/test_certs/server/key.pem \
+-CAfile /home/vagrant/php-amqp/provision/test_certs/testca/cacert.pem
+
+openssl s_client -connect localhost:8443 \
+-cert /home/vagrant/php-amqp/provision/test_certs/sasl-client/cert.pem \
+-key /home/vagrant/php-amqp/provision/test_certs/sasl-client/key.pem \
+-CAfile /home/vagrant/php-amqp/provision/test_certs/testca/cacert.pem
+
+# to connect to rabbitmq server manually
+openssl s_client -connect localhost:5671 \
+-cert /home/vagrant/php-amqp/provision/test_certs/sasl-client/cert.pem \
+-key /home/vagrant/php-amqp/provision/test_certs/sasl-client/key.pem \
+-CAfile /home/vagrant/php-amqp/provision/test_certs/testca/cacert.pem
+
+Certificate authentication needs the hostname to be as the certificate CN one:
+e.g.: server CN = *.example.org
+=> hostname must be in same domain
+e.g. rabbitmq.example.org
+if verify = true
+=> you need a /etc/hosts entry, so that the server can resolve the name supplied by the client
+127.0.0.1   rabbitmq.example.org rabbitmq
+CN in client certificate must also match CN scheme
+CN in server is *.example.org
+CN in clients used are e.g.:
+    client.example.org
+    sasl-client.example.org
+users have to exist with appropriate rights in rabbitmq
+you can validate authentication by looking at /var/log/rabbitmq/rabbit@vagrant-sasl.log
+connection <0.1104.0> (127.0.0.1:48898 -> 127.0.0.1:5671): user 'sasl-client.example.org' authenticated and granted access to vhost '/'
+*/
+$credentials = array(
+    'host' => 'rabbitmq.example.org',
+    'port' => 5671,
+
+    'verify' => true,
+
+    'cacert' => "/home/vagrant/php-amqp/provision/test_certs/testca/cacert.pem",
```

Comment:
(a note for myself) let's not using absolute route here. I guess sth like `__DIR__ . '../provision/....'` should do the job pretty well. This probably could/should be handled during further refactoring steps.